### PR TITLE
Basic German Translation

### PR DESCRIPTION
**Pull request by alanmcruickshank**
Not a full German translation but most of the core interface. Updated the `BABEL_DEFAULT_FOLDER` config flag which appeared to be pointing at the wrong folder.

Update the standard messages files in the process.

### CODE REVIEW

**Comment by xrmx** [comment]
Puzzled on why you are updating all the other languages compiled translations, and please rebase :)

**Comment by alanmcruickshank** [comment]
Rebased and conflicts resolved.

File: superset/config.py [hunk 1]
```diff
@@ -154,14 +154,15 @@
 # Setup default language
 BABEL_DEFAULT_LOCALE = 'en'
 # Your application default translation path
-BABEL_DEFAULT_FOLDER = 'babel/translations'
+BABEL_DEFAULT_FOLDER = 'superset/translations'
```
Comment (mistercrunch) [inline superset/config.py]: I'm confused and wondering how it worked before or what broke this.

Comment (alanmcruickshank) [reply superset/config.py]: Yeah I thought the same. It took me a while to set up my environment and this was one thing I changed of a few to get things working - so this change is not necessary. That being said there isn't a `babel/translations` path that I could see.

I've just done a quick test and the translations work with either value.

Vanilla FAB has the path as just `translations`
https://github.com/dpgaspar/Flask-AppBuilder/blob/af095ac4094d068fa2dd467bd01fa8b95d697fc1/examples/quickfiles/config.py

I suspect in our docs we always get people to specify the path as `superset/translations` in all the scripts for generating translations so even if the value we had before wasn't correct we were always overwriting it. I've done some quick testing and I actually suspect in most cases this line isn't used.

Comment (alanmcruickshank) [reply superset/config.py]: That being said - given it doesn't matter. I'm happy to just leave it as it is if you'd rather.

**Comment by mistercrunch** [comment]
Will merge upon conflict resolved

**Comment by alanmcruickshank** [comment]
@mistercrunch - I've removed the edits to messages.pot as they're superceded by the translation consolidation in #3750. Rebased onto current master.

**Pull request merged**